### PR TITLE
Pictogram Search (#46)

### LINKED ISSUE #46: As a guardian I would like to be able to search for pictograms when adding a new activity so that I can find a pictogram that suits the activity

### PR DESCRIPTION
**Pull request by boginw**
This closes #46

### CODE REVIEW

**Comment by TobiasPalludan** [comment]
@Tricky12321 and @ralled96 would you review this?
Here's the [review checklist](https://github.com/aau-giraf/wiki/blob/master/process_manual/code_workflow.md#code-review)

File: pubspec.yaml [hunk 1]
```diff
@@ -11,12 +11,16 @@ dependencies:
...
+  mockito: ^4.0.0
+
 
   flutter:
     sdk: flutter
   cupertino_icons: ^0.1.2
 
 dev_dependencies:
+  async_test:
+    git: git@github.com:ptope/async_test.git
```
Comment (boisenme) [inline pubspec.yaml]: This returns an error when trying to get packages

`"Could not find a file named "pubspec.yaml" in git@https://github.com/ptope/async_test 8661d8aecd626f7f57ccbcb735553edc05a2e713."`

A simple fix would be to use this instead
```
    git: 
      url: https://github.com/ptope/async_test
```

Comment (boginw) [reply pubspec.yaml]: This has been fixed

**Review by thomaslhansen — DISMISSED** [review]
We should setup some protection with regards to search time. I just ran the app locally and the API crashed doe to an flood of http requests. Perhaps @aau-giraf/sw610f19 will make an issue out of this. I do not know how this could be fixed.

**Comment by Tekrus** [comment]
@thomaslhansen We can make an issue out of it - it depends on whether the group has an idea for a fix that can be easily implemented in this PR

**Comment by boginw** [comment]
> @thomaslhansen We can make an issue out of it - it depends on whether the group has an idea for a fix that can be easily implemented in this PR

A fix could be implemented here, but I see it as a Back-end and API-client issue. It seems that the back-end is unable to keep up with the requests sent.

**Comment by Tekrus** [comment]
> 
> 
> > @thomaslhansen We can make an issue out of it - it depends on whether the group has an idea for a fix that can be easily implemented in this PR
> 
> A fix could be implemented here, but I see it as a Back-end and API-client issue. It seems that the back-end is unable to keep up with the requests sent.

Agreed. This is now an issue in the API repo:
https://github.com/aau-giraf/web-api/issues/7

**Review by ralled96 — DISMISSED** [review]

**Review by MStenkaer — DISMISSED** [review]
I have found a bug, that the pictograms change as you scroll the search results. To recreate the bug, search for "kat" wait for it to be done loading. Delete "t" and try to scroll the view.

The bug can be seen in the attached video:
https://drive.google.com/file/d/16bleafzDT7-jhNdEEvLYcT3CfUTdp2g9/view?usp=sharing

Additionally the pictogramSearch crashes sometimes following the bug described before, but this cannot be recreated.

File: lib/screens/pictogram_search_screen.dart [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import 'package:weekplanner/widgets/giraf_app_bar_widget.dart';
+import 'package:weekplanner/widgets/pictogram_image.dart';
+
+class PictogramSearch extends StatelessWidget {
+  final PictogramBloc bloc = di.getDependency<PictogramBloc>();
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: GirafAppBar(title: "Pictogram"),
```
Comment (Tekrus) [inline lib/screens/pictogram_search_screen.dart]: Title should probably in Danish (piktogram)

**Review by Tekrus — CHANGES_REQUESTED** [review]

**Review by Tekrus — DISMISSED** [review]

**Review by ralled96 — APPROVED** [review]

**Review by Tekrus — APPROVED** [review]

**Comment by boginw** [comment]
@Moesgaarda only the author should merge!

**Pull request merged**